### PR TITLE
network - for infura networks use the infura block tracker provider

### PR DESCRIPTION
**Pull request by kumavis**
This retargets our block poller for infura-based networks to their RESTful CDN endpoint. Less load for them, hopefully better response times globally.

works normally for other rpc-based network

relevant dep changes:
https://github.com/MetaMask/provider-engine/commit/c25156d618a1d9712d5f401ab67b6e93fbe3dddb
https://github.com/kumavis/eth-json-rpc-infura

### CODE REVIEW

File: app/scripts/controllers/network.js
Comment (frankiebee) [inline app/scripts/controllers/network.js]: seems like something like this belongs in the config file

File: package.json
Comment (frankiebee) [inline package.json]: i would be willing to merge this except the test are failing

**Review by frankiebee — DISMISSED** [review]

**Comment by frankiebee** [comment]
other then the test failing in `web3-provider-engine` it looks good and seems to be working in initial QA and the `eth-json-rpc-infura` looks good. It also looks like some of the tests failing are  solc tests so not sure what has changed to make those test fail when they were passing in 13.3.3

maybe @danfinlay has a better idea? if this looks good to you then i dont mind merging it.

**Comment by tmashuang** [comment]
Getting `ReferenceError: err is not defined` from `createAsyncMiddleware` that is wrapped in `createInfuraMiddleware`. Error does not happen consistently, but reproduced it 2 out of 5 times while  doing a run through.

**Comment by frankiebee** [comment]
@tmashuang can i get a QA pass from you relating to this code?

**Review by frankiebee — APPROVED** [review]

**Comment by tmashuang** [comment]
Good to merge, but will introduce #2742 as an edge case will need to expand on validation for successful resubmit.

**Review by frankiebee — COMMENTED** [review]
will merge after this is merged/fixed: https://github.com/kumavis/eth-json-rpc-infura/pull/2

**Pull request merged**
